src/webapp/reports/autogenerated-forms/DisaggregatedCOCsGridViewModel.ts

```diff
@@ -0,0 +1,190 @@
+import _ from "lodash";
+import { Section, Texts } from "../../../domain/common/entities/DataForm";
+import { DataFormInfo } from "./AutogeneratedForm";
+import { DataElement } from "../../../domain/common/entities/DataElement";
+import { Indicator } from "../../../domain/common/entities/Indicator";
+import { getDescription } from "../../../utils/viewTypes";
+import { Maybe } from "../../../utils/ts-utils";
+
+export class DisaggregatedCOCsGridViewModel {
+    static get(section: Section, dataFormInfo: DataFormInfo): Grid {
+        const columns = DisaggregatedCOCsGridViewModel.getGridColumns(section, dataFormInfo);
+        const rows = DisaggregatedCOCsGridViewModel.getGridRows(section, columns);
+        const summary = DisaggregatedCOCsGridViewModel.getSummary(section, dataFormInfo, columns);
+
+        return {
+            id: section.id,
+            indicators: section.indicators,
+            name: section.name,
+            columns: columns,
+            rows: rows,
+            toggle: section.toggle,
+            toggleMultiple: section.toggleMultiple,
+            texts: section.texts,
+            summary: section.totals ? summary : [],
+        };
+    }
+
+    private static getGridColumns(section: Section, dataFormInfo: DataFormInfo): Column[] {
+        const columns = section.dataElements.map(dataElement => {
+            const columnItems = _(dataElement.categoryOptionCombos)
+                .map(coc => {
+                    const columnName = getColumnNameFromCoc(coc) || "";
+
+                    return {
+                        name: columnName || "",
+                        description: getDescription(section.columnsDescriptions, dataFormInfo, columnName),
+                    };
+                })
+                .uniqBy("name")
+                .value();
+
+            return { columnItems: sortItems(columnItems), dataElement: dataElement };
+        });
+
+        return columns;
+    }
+
+    private static getGridRows(section: Section, columns: Column[]): Row[] {
+        const rows = _(section.dataElements)
+            .flatMap(dataElement => dataElement.categoryOptionCombos)
+            .groupBy(coc => getRowNameFromCoc(coc))
+            .map((_, rowName) => {
```

Comment:
`rowName` can be `undefined` because getRowNameFromCoc return `Maybe<string>`, Should we filter undefined using compact or at least log log using console.warn? I think it could be weird to have a rowName empty or undefined.